base/services/w32time/W32Time.c

```diff
@@ -0,0 +1,370 @@
+        {
+            break;
+        }
+    }
+
+
+    dwIndex = 0;
+    while (TRUE)
+    {
+        dwValSize = MAX_VALUE_NAME * sizeof(WCHAR);
+        szValName[0] = L'\0';
+        lRet = RegEnumValueW(hKey,
+                             dwIndex,
+                             szValName,
+                             &dwValSize,
+                             NULL,
+                             NULL,
+                             (LPBYTE)szData,
+                             &dwNameSize);
+        if (lRet == ERROR_SUCCESS)
+        {
+            /* Get date from selected reg value */
+            if (wcscmp(szValName, szDefault) == 0) // if (Index == Default)
+            {
+                wcscpy(szDefault, szData);
+                dwIndex++;
+            }
+            else
+            {
+                dwIndex++;
+            }
+        }
+        else if (lRet != ERROR_MORE_DATA)
+        {
+            break;
+        }
+    }
+
+    RegCloseKey(hKey);
+
+    ulTime = GetServerTime(szDefault);
+
+    if (ulTime != 0)
+        UpdateSystemTime(ulTime);
+
+    return 0;
+}
+
+static VOID CALLBACK
+ServiceMain(int argc, char** argv) 
+{
+    int   error;
+    int   result;
+    DWORD dwPollInterval;
+ 
+    UNREFERENCED_PARAMETER(argc);
+    UNREFERENCED_PARAMETER(argv);
+
+    ServiceStatus.dwServiceType             = SERVICE_WIN32; 
+    ServiceStatus.dwCurrentState            = SERVICE_START_PENDING; 
+    ServiceStatus.dwControlsAccepted        = SERVICE_ACCEPT_STOP | SERVICE_ACCEPT_SHUTDOWN;
+    ServiceStatus.dwWin32ExitCode           = 0; 
+    ServiceStatus.dwServiceSpecificExitCode = 0; 
+    ServiceStatus.dwCheckPoint              = 0; 
+    ServiceStatus.dwWaitHint                = 0; 
+ 
+    hStatus = RegisterServiceCtrlHandler(ServiceName,
+		                         (LPHANDLER_FUNCTION)ControlHandler); 
+    if (hStatus == (SERVICE_STATUS_HANDLE)0) 
+    { 
+        // Registering Control Handler failed
+        return; 
+    }  
+    // Initialize Service 
+    error = InitService(); 
+    if (error) 
+    {
+        // Initialization failed
+        ServiceStatus.dwCurrentState       = SERVICE_STOPPED; 
+        ServiceStatus.dwWin32ExitCode      = -1; 
+        SetServiceStatus(hStatus, &ServiceStatus); 
+        return; 
+    } 
+    // We report the running status to SCM. 
+    ServiceStatus.dwCurrentState = SERVICE_RUNNING; 
+    SetServiceStatus (hStatus, &ServiceStatus);
+ 
+    // The worker loop of a service
+    while (ServiceStatus.dwCurrentState == SERVICE_RUNNING)
+        {
+            dwPollInterval = GetIntervalSetting();
+            result = set_time();
+            if (result)
+                {
+                    ServiceStatus.dwCurrentState       = SERVICE_STOPPED; 
+                    ServiceStatus.dwWin32ExitCode      = -1; 
+                    SetServiceStatus(hStatus, &ServiceStatus);
+                    return;
+                }
+            Sleep(dwPollInterval);
+        }
```

Comment:
Broken indenting

Reply:
I will move the upper braces and their contents back 4 spaces.